src/quo2/components/dropdown.cljs

```diff
@@ -0,0 +1,150 @@
+                                               (if (swap! open? not)
+                                                 (apply-anim dd-height 120)
+                                                 (apply-anim dd-height 0)))
+                                   :style (cond-> (merge
+                                                   (if icon
+                                                     padding-with-icon
+                                                     padding-with-no-icon)
+                                                   {:width (if icon
+                                                             width-with-icon
+                                                             width)
+                                                    :height height
+                                                    :border-radius (if (= :small size)
+                                                                     8
+                                                                     10)
+                                                    :flex-direction :row
+                                                    :align-items :center
+                                                    :border-width 1
+                                                    :border-color (if @open?
+                                                                    border-color
+                                                                    (color-by-10 border-color))
+                                                    :background-color (if @open?
+                                                                        dd-color
+                                                                        (color-by-10 dd-color))})
+                                            disabled? (assoc :opacity 0.3))}
+                            disabled? (assoc :disabled true))
+     (when icon
+       [icons/icon icon {:no-color true
+                         :size 20
+                         :container-style {:margin-right spacing
+                                           :margin-top 1
+                                           :width icon-size
+                                           :height icon-size}}])
+     [text/text {:size font-size
+                 :weight :medium
+                 :font :font-medium
+                 :color :main
+                 :style {:margin-right spacing}} text]
+     [icons/icon (if @open?
+                   (if dark?
+                     :main-icons/pullup
+                     :main-icons/pullup-black)
+                   (if dark?
+                     :main-icons/dropdown-white
+                     :main-icons/dropdown-black))
+      {:no-color true
+       :size 20
+       :container-style {:width icon-size
+                         :margin-top 1
+                         :height icon-size}}]]))
+
+(defn items-comp
+  [{:keys [items on-select]}]
+  (let [items-count (count items)]
+    [:f>
```

Comment:
Quick question: Why do we need `:f>` and the form-2 component (with the anonymous `fn` macro)? Given that the component doesn't have local state, couldn't this component be a simple form-1?

Reply:
To create a functional component

Reply:
I tried without it, And the functional component didn't work .. Maybe someone from the team can help out
@flexsurfer @Parveshdhull

Reply:
Thanks for taking a look @ibrkhalil. I tested this branch locally, but the `items-comp` function works well with/without the surrounding functional form. That is, I can select the elements in the dropdown, the animations run and the state is correctly updated. What exactly did you find that's not working?

By the way, while checking this I noticed the dropdown is not displaying the currently active element. The diff below fixes the issue (not tested at length, but seems to be working):

```diff
modified   src/quo2/components/dropdown.cljs
@@ -40,7 +40,7 @@
   [color]
   (colors/alpha color 0.6))
 
-(defn dropdown-comp [{:keys [icon open? dd-height size disabled? dd-color border-color]}]
+(defn dropdown-comp [{:keys [icon open? dd-height size disabled? dd-color border-color default-item]}]
   (let [dark?                                                         (colors/dark?)
         {:keys [width height width-with-icon padding font icon-size]} (size sizes)
         {:keys [padding-with-icon padding-with-no-icon]}              padding
@@ -86,7 +86,7 @@
      [text/text {:size   font-size
                  :weight :medium
                  :font   :font-medium
-                 :color  :main} "Dropdown"]
+                 :color  :main} default-item]
      [icons/icon (if @open?
                    (if dark?
                      :main-icons/pullup-dark
```

Reply:
Thanks Icaro for taking a look!
Actually the design request is for the dropdown element to always show Dropdown on it's view

I'll make a video of the issue I get when removing the functional component usage

Reply:
> Actually the design request is for the dropdown element to always show Dropdown on it's view

Oh I missed that. Thanks @ibrkhalil

Reply:
https://user-images.githubusercontent.com/33176106/190894785-44c0011d-339b-4a98-ac4e-99d6712e82be.mp4

Here's the video Icaro
And also I found this for [reference](https://clojurians-log.clojureverse.org/reagent/2021-02-19/1613740755.079900)

Reply:
Thanks for the detailed video @ibrkhalil